src/webAppCreator.ts

```diff
@@ -0,0 +1,481 @@
+        }
+
+        return subscriptionStep.subscription;
+    }
+
+    protected getSelectedResourceGroup(): ResourceModels.ResourceGroup {
+        const resourceGroupStep = <ResourceGroupStep>this.wizard.findStep(step => step instanceof ResourceGroupStep, 'The Wizard must have a ResourceGroupStep.');
+
+        if (!resourceGroupStep.resourceGroup) {
+            throw new Error('A resource group must be selected first.');
+        }
+
+        return resourceGroupStep.resourceGroup;
+    }
+}
+
+class SubscriptionStep extends WizardStep {
+    private _subscription: SubscriptionModels.Subscription;
+
+    constructor(wizard: WizardBase, readonly azureAccount: AzureAccountWrapper) {
+        super(wizard, 'Select subscription');
+    }
+
+    async prompt(): Promise<void> {
+        const inFilterSubscriptions = await this.azureAccount.getFilteredSubscriptions();
+        const otherSubscriptions = await this.azureAccount.getAllSubscriptions();
+        const quickPickItems: vscode.QuickPickItem[] = [];
+        const quickPickOptions = { placeHolder: `Select the subscription where the new Web App will be created in. (${this.stepProgressText})` };
+
+        inFilterSubscriptions.forEach(s => {
+            const index = otherSubscriptions.findIndex(other => other.subscriptionId === s.subscriptionId);
+            if (index >= 0) {   // Remove duplicated items from "all subscriptions".
+                otherSubscriptions.splice(index, 1);
+            }
+
+            const item = {
+                label: `📌 ${s.displayName}`,
+                description: '',
+                detail: s.subscriptionId
+            };
+
+            quickPickItems.push(item);
+        });
+
+        otherSubscriptions.forEach(s => {
+            const item = {
+                label: s.displayName,
+                description: '',
+                detail: s.subscriptionId
+            };
+
+            quickPickItems.push(item);
+        });
+
+        const result = await this.showQuickPick(quickPickItems, quickPickOptions);
+        this._subscription = inFilterSubscriptions.concat(otherSubscriptions).find(s => s.subscriptionId === result.detail);
+    }
+
+    async execute(): Promise<void> {
+        this.wizard.writeline(`The new Web App will be created in subscription "${this.subscription.displayName}" (${this.subscription.subscriptionId}).`);
+    }
+
+    get subscription(): SubscriptionModels.Subscription {
+        return this._subscription;
+    }
+}
+
+class ResourceGroupStep extends SubscriptionBasedWizardStep {
+    private _createNew: boolean;
+    private _rg: ResourceModels.ResourceGroup;
+
+    constructor(wizard: WizardBase, azureAccount: AzureAccountWrapper) {
+        super(wizard, 'Select or create resource group', azureAccount);
+    }
+
+    async prompt(): Promise<void> {
+        const createNewItem: vscode.QuickPickItem = {
+            label: '+ New Resource Group',
```

Comment:
I meant to literally use `$(plus)`. It displays as a larger plus icon and looks better than the `+` character

Reply:
And I believe 'Create' is used more often than 'New' in VS Code